2015/proposals/banaszkiewicz-piotr.md

```diff
@@ -0,0 +1,158 @@
+# Installation-test scripts: aggregation server, script enhancements
+
+## Abstract
+
+While Software Carpentry workshops gather a lot of feedback regarding
+instructors' teaching, not much attention was drawn to students' computers and
+their issues with installation of open source software used during workshops.
+
+This project will focus on bringing a working server for aggregating of
+Software Carpentry installation testing scripts with additional goal of
+providing easier and better experience for students using the script itself.
+
+## Features
+
+The server should be able to aggregate incoming (via REST API) data from
+installation testing script.
+
+The server should allow Software Carpentry admins to view statistics from
+uploaded data.
+
+Installation script should upload diagnostic data consisting of:
+
+* operating system information
+* installed packages and their versions
+* failed packages checks and error messanges
+
+Additionally, the data may be matched against workshop
+[reference package list](https://github.com/wking/swc-setup-installation-test/issues/2).
+
+## Technical Details
+
+The server will be implemented using [Django](https://www.djangoproject.com/)
+and [SQLite](https://docs.djangoproject.com/en/1.7/ref/databases/) database
+for lower footprint and easier maintainance (and because I feel comfortable
+using them).
+
+API, in a RESTful fashion, will be implemented using
+[django-tastypie](http://tastypieapi.org/).
+
+Rough database structure:
+
+* system information table
+    * operating system family (Windows / MacOSX / Linux / Other)
+    * OS version (8.1 / 10.9 / Ubuntu / ?)
+    * Hardware architecture (x86_64 / x86 / Other)
```

Comment:
How is "Other" going to work?  This information will mostly be coming straight from the Python checker, so you can just accept the arch values Python detects.  Or just have it be a user- (checker-script-) submitted string.  I'd avoid actually storing “Other” or NULLs in the database for this field.